shell/platform/linux/fl_text_input_plugin.cc

```diff
@@ -138,9 +165,42 @@ static void perform_action(FlTextInputPlugin* self) {
                                   nullptr, perform_action_response_cb, self);
 }
 
+// Signal handler for GtkIMContext::preedit-start
+static void im_preedit_start_cb(FlTextInputPlugin* self) {
+  self->text_model->BeginComposing();
+
+  // Set the top-level window used for system input method windows.
+  GdkWindow* window =
+      gtk_widget_get_window(gtk_widget_get_toplevel(GTK_WIDGET(self->view)));
+  gtk_im_context_set_client_window(self->im_context, window);
+}
+
+// Signal handler for GtkIMContext::preedit-changed
+static void im_preedit_changed_cb(FlTextInputPlugin* self) {
+  gchar* buf = nullptr;
```

Comment:
Thus should be `g_autofree gchar* buf = nullptr;` - buf is currently leaking if `UpdateComposingText` doesn't take ownership of it.

Reply:
Oh yikes. Good catch. Thanks.